randomGuess.js

```diff
@@ -0,0 +1,95 @@
+const readline = require('readline');
+
+const rl = readline.createInterface(process.stdin, process.stdout);
+
+function ask(questionText) {
+	return new Promise((resolve, reject) => {
+		rl.question(questionText, resolve);
+	});
+}
+
+async function guessFcn(maxGuess, minGuess) {
+  //funciton takes the changing min and max and makes a random guess within the parameters
+	const maxNumber = Number.parseFloat(maxGuess);
+
+	const minNumber = Number.parseFloat(minGuess);
+
+	const difference = maxNumber - minNumber;
+
+	const random = Math.random();
+
+	const product = Number.parseFloat(random * difference);
+
+	const rawGuess = Number(product) + Number(minGuess + 1);
+
+	let guess = Math.floor(rawGuess);
+
+	let response = await ask('Is it ' + guess + '? (yes or no)  ');
+
+	//sanitize response
+
+	let responseSan = response.toLowerCase().trim();
+
+	if (responseSan === 'yes') {
+		console.log('Conratulations you have guessed the number.');
+
+		process.exit();
+	}
+
+	if (responseSan === 'no') {
+    //triggers if no response giving the computer guidance
+		await ifNo(guess, maxNumber, minNumber);
+	}
+
+	if (responseSan !== 'no' || responseSan !== 'yes') {
+    // repeats question when recieving unrecognized responses
+		await guessFcn(maxGuess, minGuess);
```

Comment:
This asks the question here but when you enter a response that isn't recognized the computer asks a new number. This is an example of using recursion which makes your code harder to debug than a function with just a while loop, which you could use to achieve the same effect that you are going for here.
```js
while(userResponse !== expectedResponse) {
    userResponse = await ask('please enter a valid response')
}
```

Reply:
Also this sort of traps you in a not quite infinite loop. It is an infinite loop, but since you are awaiting and reassigning inside of it you can never move on to higher/lower because `responseSan !== 'no' || responseSan !== 'yes'` will always be true. You should use use the logical and opertaor `&&` here instead of the logical or operator.

Reply:
So it's not the recursion that's causing the bug, it's actually the fact that this logic check is always true. Use `&&` instead of `||`